Library/Homebrew/brew.sh

```diff
@@ -3,15 +3,27 @@
 ##### to be able to `source` in shell configurations run quickly.
 #####
 
-# Doesn't need a default case because we don't support other OSs
+# Doesn't need a default case because we don't support other OSs/architectures
 # shellcheck disable=SC2249
-HOMEBREW_PROCESSOR="$(uname -m)"
-HOMEBREW_PHYSICAL_PROCESSOR="${HOMEBREW_PROCESSOR}"
-HOMEBREW_SYSTEM="$(uname -s)"
-case "${HOMEBREW_SYSTEM}" in
-  Darwin) HOMEBREW_MACOS="1" ;;
-  Linux) HOMEBREW_LINUX="1" ;;
+case "${MACHTYPE}" in
+  arm64-*)
+    HOMEBREW_PROCESSOR="arm64"
+    ;;
+  x86_64-*)
+    HOMEBREW_PROCESSOR="x86_64"
+    ;;
+esac
```

Comment:
`aarch64` is probably worth adding (for Linux), along with a generic fallback to `uname -m` if nothing matches.

Homebrew does have support for PPC and 32-bit arm.

Reply:
Not gonna do the performance optimisation case for `aarch64` when it's very much unsupported but have added the fallback.